pkg/devfile/adapters/kubernetes/component/adapter.go

```diff
@@ -101,79 +93,15 @@ func (a Adapter) Push(parameters common.PushParameters) (err error) {
-		if componentExists {
-			spinner.Start("Checking file changes for pushing", false)
-		} else {
-			// if the component doesn't exist, we don't check for changes in the files
-			// thus we show a different message
-			spinner.Start("Checking files for pushing", false)
-		}
-
-		// Before running the indexer, make sure the .odo folder exists (or else the index file will not get created)
-		odoFolder := filepath.Join(parameters.Path, ".odo")
-		if _, err := os.Stat(odoFolder); os.IsNotExist(err) {
-			err = os.Mkdir(odoFolder, 0750)
-			if err != nil {
-				return errors.Wrap(err, "unable to create directory")
-			}
-		}
-
-		// run the indexer and find the modified/added/deleted/renamed files
-		filesChanged, filesDeleted, err := util.RunIndexer(parameters.Path, absIgnoreRules)
-		spinner.End(true)
-
-		if err != nil {
-			return errors.Wrap(err, "unable to run indexer")
-		}
-
-		// If the component already exists, sync only the files that changed
-		if componentExists {
-			// apply the glob rules from the .gitignore/.odo file
-			// and ignore the files on which the rules apply and filter them out
-			filesChangedFiltered, filesDeletedFiltered := util.FilterIgnores(filesChanged, filesDeleted, absIgnoreRules)
-
-			// Remove the relative file directory from the list of deleted files
-			// in order to make the changes correctly within the Kubernetes pod
-			deletedFiles, err = util.RemoveRelativePathFromFiles(filesDeletedFiltered, parameters.Path)
-			if err != nil {
-				return errors.Wrap(err, "unable to remove relative path from list of changed/deleted files")
-			}
-			glog.V(4).Infof("List of files to be deleted: +%v", deletedFiles)
-			changedFiles = filesChangedFiltered
-
-			if len(filesChangedFiltered) == 0 && len(filesDeletedFiltered) == 0 {
-				// no file was modified/added/deleted/renamed, thus return without building
-				log.Success("No file changes detected, skipping build. Use the '-f' flag to force the build.")
-				return nil
-			}
-		}
-	} else if len(parameters.WatchFiles) > 0 || len(parameters.WatchDeletedFiles) > 0 {
-		changedFiles = parameters.WatchFiles
-		deletedFiles = parameters.WatchDeletedFiles
-	}
-
-	if parameters.ForceBuild || !componentExists || podChanged {
-		isForcePush = true
+	// Find at least one pod with the source volume mounted, error out if none can be found
+	containerName, err := getFirstContainerWithSourceVolume(pod.Spec.Containers)
+	if err != nil {
+		return errors.Wrapf(err, "error while retrieving container from pod: %s", podName)
 	}
 
-	// Sync the local source code to the component
-	err = a.pushLocal(parameters.Path,
-		changedFiles,
-		deletedFiles,
-		isForcePush,
-		globExps,
-		pod.GetName(),
-		pod.Spec.Containers,
-	)
+	// Get a sync adapter. Check if project files have changed and sync accordingly
+	syncAdapter := sync.New(a.AdapterContext, &a.Client)
+	err = syncAdapter.CheckProjectFiles(parameters, pod.GetName(), containerName, podChanged, componentExists)
```

Comment:
Can we change the name of `CheckProjectFiles` to be something more clear? Something like `SyncFiles`?